dace/sdfg/utils.py

```diff
@@ -2116,3 +2116,192 @@ def set_nested_sdfg_parent_references(sdfg: SDFG):
             if isinstance(node, NestedSDFG):
                 node.sdfg.parent_sdfg = sdfg
                 set_nested_sdfg_parent_references(node.sdfg)
+
+
+def get_used_data(scope: Union[ControlFlowRegion, SDFGState, nd.MapEntry, nd.NestedSDFG],
```

Comment:
This screams analysis pass! :) I'm sure this would be useful as cached information in the pass pipeline infrastructure, because this kind of information is often used in other places too.